### PR TITLE
[connect-udp] Add missing error reasons for H2O_HTTP3_ERROR_GENERAL_PROTOCOL

### PR DESCRIPTION
**Pull request by deweerdt**

### CODE REVIEW

File: lib/common/http3client.c [hunk 1]
```diff
@@ -504,21 +504,28 @@ static int handle_input_expect_headers(struct st_h2o_http3client_req_t *req, con
             return 0;
         }
         notify_response_error(req, *err_desc != NULL ? *err_desc : "qpack error");
+        if (*err_desc == NULL) {
+            *err_desc = "qpack error";
+        }
```
Comment (kazuho) [inline lib/common/http3client.c:509]:
```suggestion
        if (*err_desc == NULL)
            *err_desc = "qpack error";
        notify_response_error(req, *err_desc);
```
I think we can be less redundant?

File: lib/common/http3client.c [hunk 2]
```diff
@@ -504,21 +504,28 @@ static int handle_input_expect_headers(struct st_h2o_http3client_req_t *req, con
...
         return H2O_HTTP3_ERROR_GENERAL_PROTOCOL; /* FIXME */
     }
     if (header_ack_len != 0)
         h2o_http3_send_qpack_header_ack(&req->conn->super, header_ack, header_ack_len);
 
     if (datagram_flow_id.base != NULL) {
-        if (!req->offered_datagram_flow_id)
+        if (!req->offered_datagram_flow_id) {
+            *err_desc = "No offered datagram flow id";
```
Comment (kazuho) [inline lib/common/http3client.c:517]:
```suggestion
            *err_desc = "no offered datagram-flow-id";
```
"datagram-flow-id" is a header field name using hyphens. Our convention is to start with a lowercase letter.

File: lib/common/http3client.c [hunk 3]
```diff
@@ -504,21 +504,28 @@ static int handle_input_expect_headers(struct st_h2o_http3client_req_t *req, con
...
         /* TODO validate the returned value */
     }
 
     /* handle 1xx */
     if (100 <= status && status <= 199) {
         if (status == 101) {
-            notify_response_error(req, "unexpected 101");
+            static const char unexpected_101[] = "unexpected 101";
+            notify_response_error(req, unexpected_101);
+            *err_desc = unexpected_101;
```
Comment (kazuho) [inline lib/common/http3client.c:528]:
```suggestion
            *err_desc = "unexpected 101";
            notify_response_error(req, *err_desc);
```
Take-it-or-leave-it, but if we want to use the convention of setting `*err_desc` to appropriate raw string literal then use it, maybe.

File: lib/http3/server.c [hunk 4]
```diff
@@ -1235,8 +1237,10 @@ static int handle_input_expect_headers(struct st_h2o_http3_server_stream_t *stre
         return handle_input_expect_headers_send_http_error(stream, h2o_send_error_400, "Invalid Request", *err_desc, err_desc);
 
     /* validate semantic requirement */
-    if (!h2o_req_validate_pseudo_headers(&stream->req))
+    if (!h2o_req_validate_pseudo_headers(&stream->req)) {
+        *err_desc = "Invalid pseudo headers";
```
Comment (kazuho) [inline lib/http3/server.c:1241]:
```suggestion
        *err_desc = "invalid pseudo headers";
```

**Review by kazuho — COMMENTED** [review]
Thank you for the PR. Looks good, I only have nitpicks.

**Comment by deweerdt** [comment]
Thank you @kazuho ! Applied all.

**Pull request merged**